lib/modules/manager/cargo/extract.spec.ts

```diff
@@ -112,6 +113,68 @@ describe('modules/manager/cargo/extract', () => {
       expect(res?.deps).toHaveLength(3);
     });
 
+    it('extracts workspace dependencies', async () => {
+      const cargoToml = codeBlock`
+[package]
+name = "renovate-test"
+version = "0.1.0"
+authors = ["John Doe <john.doe@example.org>"]
+edition = "2018"
+
+[dependencies]
+git2 = "0.14.0"
+
+[workspace]
+members = ["pcap-sys"]
+
+[workspace.dependencies]
+serde = "1.0.146"
+tokio = { version = "1.21.1" }`;
+      const res = await extractPackageFile(cargoToml, 'Cargo.toml', config);
+      expect(res?.deps).toEqual([
+        {
+          currentValue: '0.14.0',
+          datasource: 'crate',
+          depName: 'git2',
+          depType: 'dependencies',
+          managerData: { nestedVersion: false },
+        },
+        {
+          currentValue: '1.0.146',
+          datasource: 'crate',
+          depName: 'serde',
+          depType: 'dependencies',
```

Comment:
`workspace.dependencies` can be used by any `depType`.  Should these get their own `depType`?

(note: I was the cargo team member who mentored `workspace.dependencies` support)

Reply:
Good question. I guess the current deptype is too narrow for workspace.dependencies. it would probably be for the best. As you say workspace.deps, can be used by all types. However a bit problematic is that they exactly can be used by the different depTypes. But how do we differentiate them? The client should probably have the option to target the different devTypes independently.

However, just having a depType called workspace.dependencies or something would probably be good enough for now. 

It just needs to be included by the default preset to also be targeted

Reply:
In my ideal world (i.e. what I might do for `cargo-upgrade`) is if a dependency is only used for one depType, then I will classify it as that depType.  Unsure if that is possible to do with renovate's architecture.

Reply:
> In my ideal world (i.e. what I might do for `cargo-upgrade`) is if a dependency is only used for one depType, then I will classify it as that depType.  Unsure if that is possible to do with renovate's architecture.

that's not easily possible and makes code unnecessary complex. so I suggest to use another dep type like suggested.